app/src/org/commcare/tasks/SendTask.java

```diff
@@ -91,6 +94,26 @@ protected Boolean doTaskBackground(Void... params) {
                 continue;
             }
             try {
+
+                // see if we have a form.properties file to load the PostURL from
+                FilenameFilter filter = new FilenameFilter() {
+                    @Override
+                    public boolean accept(File dir, String filename) {
+                        return filename.equals(ZipTask.FORM_PROPERTIES_FILE);
+                    }
+                };
+
+                File[] fileContents = f.listFiles(filter);
+                if(fileContents != null && fileContents.length > 0){
+                    Properties properties = FileUtil.loadProperties(fileContents[0]);
+                    if(properties != null && properties.getProperty(ZipTask.FORM_PROPERTY_POST_URL) != null){
+                        url = properties.getProperty(ZipTask.FORM_PROPERTY_POST_URL);
+                        Logger.log(TAG, "Successfully got form.property PostURL: " + url);
+                    }
+                    // don't submit this file
+                    FileUtil.deleteFileOrDir(fileContents[0]);
+                }

```

Comment:
1) I'm okay with helper methods changing an object's state (wasn't necessarily advocating that the method should be `static`).
2) I like to use helper methods to help mitigate the code readability impact of code growth, so in my opinion lifting this code into helper method(s) is a way to be proactive about the fact that more properties may get added later.